Added a progress bar for upload and download of object to a bucket.

anvithks opened the pull request:
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
This PR adds a progress bar to show the progress of upload and download of an object to bucket.


**Test Report Added?**:
/kind TESTED


**Test Report**:
**Upload Progress**

![upload-progress](https://user-images.githubusercontent.com/19162717/85223126-066ade00-b3de-11ea-8127-0408284425c6.png)

**Download Progress**
![download-progress](https://user-images.githubusercontent.com/19162717/85223129-08cd3800-b3de-11ea-8fbe-ba9ed8a0f1fd.png)

**Special notes for your reviewer**:

kumarashit commented on src/app/business/block/bucket-detail/bucket-detail.component.ts:
```diff
@@ -151,42 +154,42 @@ export class BucketDetailComponent implements OnInit {
-        }, (error)=>{
-          window.sessionStorage['searchIndex'] = "";
-        });
+          let url = "/" + this.bucketId + '/' + key;
```
This indentation does not look correct

kumarashit replied:
Ok...U may have to indent 156 line for four spaces

kumarashit commented:
There is already a functionality in the backend to report the progress value. I remember C2C had put this. Why don't we leverage that and show in-line than showing it at bottom

anvithks commented:
@kumarashit What C2C implemented was progress of Migration tasks. This was through the API and the progress is part of the API response of the Jobs API.
What I have implemented here is showing the upload and download progress on client side. 
I am using the PrimeNG ProgressBar module and have leveraged existing functionality provided by the `xhr.onprogress` , `xhr.upload.progress` methods.
The Upload and Download notification is existing and I have just added the progress bar to the same location. The design of the notification window can be changed. 
We can look at it at a later date.

kumarashit commented:
> @kumarashit What C2C implemented was progress of Migration tasks. This was through the API and the progress is part of the API response of the Jobs API.
> What I have implemented here is showing the upload and download progress on client side.
> I am using the PrimeNG ProgressBar module and have leveraged existing functionality provided by the `xhr.onprogress` , `xhr.upload.progress` methods.
> The Upload and Download notification is existing and I have just added the progress bar to the same location. The design of the notification window can be changed.
> We can look at it at a later date.

OK. Thanks
